pkg/upload/react-native-sourcemaps.go

```diff
@@ -0,0 +1,111 @@
+// It validates key identifiers (version name, code bundle ID, etc.) and constructs
+// upload parameters according to React Native platform conventions.
+//
+// Parameters:
+//   - globalOptions: Global CLI options including the API key and upload configuration.
+//   - logger: Logger instance for structured debug, info, and error output.
+//
+// Behavior:
+//   - Validates presence of required identifiers (versionName, versionCode/bundleVersion, codeBundleId).
+//   - Builds metadata for either iOS or Android React Native builds.
+//   - Uploads both the source map and the JS bundle to the Bugsnag /react-native-source-map endpoint.
+//
+// Returns:
+//   - error: Non-nil if validation or upload fails.
+func ProcessReactNativeSourcemaps(globalOptions options.CLI, logger log.Logger) error {
+	reactNativeOpts := globalOptions.Upload.ReactNativeSourcemaps
+	uploadOpts := make(map[string]string)
+	// Validate versioning identifiers
+	if reactNativeOpts.VersionName == "" && reactNativeOpts.CodeBundleId == "" {
+		var versionLabel string
+		switch reactNativeOpts.Platform {
+		case "android":
+			versionLabel = "version name, version code, or code bundle ID"
+		case "ios":
+			versionLabel = "version name, bundle version, or code bundle ID"
+		default:
+			versionLabel = "version name"
+		}
+
+		return fmt.Errorf(
+			"missing required identifiers: you must set at least %s to uniquely identify the build",
+			versionLabel,
+		)
+	}
+
+	// Select identification strategy
+	if reactNativeOpts.CodeBundleId != "" {
+		logger.Info(fmt.Sprintf("Using code bundle ID to identify build: %s", reactNativeOpts.CodeBundleId))
+		uploadOpts["codeBundleId"] = reactNativeOpts.CodeBundleId
+	} else {
+		logger.Info("Using version name and version code/bundle version to identify build")
+		uploadOpts["appVersion"] = reactNativeOpts.VersionName
+
+		if reactNativeOpts.VersionCode != "" {
+			logger.Debug(fmt.Sprintf("Using Android version code: %s", reactNativeOpts.VersionCode))
+			uploadOpts["appVersionCode"] = reactNativeOpts.VersionCode
+		} else if reactNativeOpts.BundleVersion != "" {
+			logger.Debug(fmt.Sprintf("Using iOS bundle version: %s", reactNativeOpts.BundleVersion))
+			uploadOpts["appBundleVersion"] = reactNativeOpts.BundleVersion
+		}
+	}
+
+	// Development mode flag
+	if reactNativeOpts.Dev {
+		logger.Debug("Development build detected — marking upload as dev=true")
+		uploadOpts["dev"] = "true"
+	}
+
+	// Determine project root
+	if reactNativeOpts.ProjectRoot == "" {
+		uploadOpts["projectRoot"] = string(reactNativeOpts.Path)
+		logger.Debug(fmt.Sprintf("Project root not provided — using inferred path: %s", uploadOpts["projectRoot"]))
+	} else {
+		uploadOpts["projectRoot"] = reactNativeOpts.ProjectRoot
+		logger.Debug(fmt.Sprintf("Using specified project root: %s", reactNativeOpts.ProjectRoot))
+	}
+
+	// Platform and overwrite flag
+	uploadOpts["platform"] = reactNativeOpts.Platform
```

Comment:
Is `Platform` a required field or could it be empty? It feels like we should validate this somewhere higher up, checking that it's one of our 3 supported ones?

Reply:
This validation has been added to the `utils` package